packages/react-magma-dom/src/components/TreeView/useTreeItem.ts

```diff
@@ -73,12 +73,21 @@ export function useTreeItem(props: UseTreeItemProps, forwardedRef) {
     initialExpandedItemsNeedUpdate,
     items,
     selectItem,
+    expandedSet,
   } = React.useContext(TreeViewContext);
 
   const treeViewItemData = React.useMemo(() => {
     return items.find(item => item.itemId === itemId);
   }, [itemId, items]);
 
+  React.useEffect(() => {
+    const isExpanded = treeViewItemData?.itemId
+      ? expandedSet.has(treeViewItemData.itemId)
+      : false;
+
+    setExpanded(isExpanded);
+  }, [expandedSet, treeViewItemData]);
```

Comment:
How is this different from line 103 `const [expanded, setExpanded] = React.useState(false);`?